File: docs/scripts/fix_imports.py
import os
import re

def fix_imports(directory):
    # Regex to find relative imports that don't have an extension
    # It looks for lines starting with import/export and having from './...' or '../...'
    import_re = re.compile(r'((import|export)\s+.+?\s+from\s+[\'"](\.?\.\/.*?))(?<!\.[a-z0-9]+)([\'"];?)')

    for root, dirs, files in os.walk(directory):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.git' in dirs:
            dirs.remove('.git')
        if '.next' in dirs:
            dirs.remove('.next')
        if 'dist' in dirs:
            dirs.remove('dist')

        for file in files:
            if file.endswith(('.ts', '.tsx')):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()

                new_content = import_re.sub(r'\1.js\4', content)

                if content != new_content:
                    print(f"Updating imports in: {path}")
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)

if __name__ == "__main__":
    project_dir = os.getcwd()
    print(f"Starting import fix in: {project_dir}")
    fix_imports(project_dir)
    print("Import fix complete.")

File: docs/scripts/fix_imports.js
import fs from 'fs';
import path from 'path';

const directory = process.cwd();

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory()) {
            if (!['node_modules', '.git', '.next', 'dist'].includes(file)) {
                walk(fullPath);
            }
        } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
            processFile(fullPath);
        }
    }
}

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Simple regex to find relative imports/exports that don't have an extension
    // Matches: import { something } from './some/path'
    // Matches: export { something } from '../another/path'
    // Captures the prefix (import/export ... from ') and the path
    const regex = /((?:import|export)\s+[\s\S]*?from\s+['"])(\.?\.\/[^'"]+)(?=['"])/g;
    
    let changed = false;
    const newContent = content.replace(regex, (match, prefix, path) => {
        // If it already has an extension, don't add .js
        if (/\.[a-z0-9]+$/i.test(path)) {
            return match;
        }
        changed = true;
        return prefix + path + '.js';
    });

    if (changed) {
        console.log(`Updated: ${filePath}`);
        fs.writeFileSync(filePath, newContent, 'utf8');
    }
}

console.log(`Starting import fix in: ${directory}`);
walk(directory);
console.log('Import fix complete.');
